File: library/src/main/java/com/proxerme/library/api/notifications/InternalApi.java
package com.proxerme.library.api.notifications;

import com.proxerme.library.api.ProxerCall;
import com.proxerme.library.entitiy.notifications.NewsArticle;
import retrofit2.http.GET;
import retrofit2.http.Query;

import java.util.List;

/**
 * @author Ruben Gees
 */
interface InternalApi {

    @GET("notifications/news")
    ProxerCall<List<NewsArticle>> news(@Query("p") Integer page, @Query("limit") Integer limit);
}

File: library/src/main/kotlin/me/proxer/library/api/wiki/InternalApi.kt
package me.proxer.library.api.wiki

import me.proxer.library.ProxerCall
import me.proxer.library.entity.wiki.WikiPage
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * @author Ruben Gees
 */
internal interface InternalApi {

    @GET("wiki/content")
    fun content(@Query("title") title: String): ProxerCall<WikiPage>
}
